model/atmosphere/advection/tests/advection_stencil_tests/test_compute_ppm_slope.py

```diff
@@ -65,12 +61,16 @@ def reference(
 
     @pytest.fixture
     def input_data(self, grid):
-        z_slope = zero_field(grid, dims.CellDim, dims.KDim)
-        p_cc = random_field(grid, dims.CellDim, dims.KDim, extend={dims.KDim: 1})
-        p_cellhgt_mc_now = random_field(grid, dims.CellDim, dims.KDim, extend={dims.KDim: 1})
+        z_slope = helpers.zero_field(grid, dims.CellDim, dims.KDim)
+        p_cc = helpers.random_field(grid, dims.CellDim, dims.KDim, extend={dims.KDim: 1})
+        p_cellhgt_mc_now = helpers.random_field(
+            grid, dims.CellDim, dims.KDim, extend={dims.KDim: 1}
+        )
         k = as_field(
             (dims.KDim,),
-            np.arange(0, _shape(grid, dims.KDim, extend={dims.KDim: 1})[0], dtype=int32),
+            np.arange(
+                0, helpers._shape(grid, dims.KDim, extend={dims.KDim: 1})[0], dtype=gtx.int32
+            ),
```

Comment:
```suggestion
            k = gtx.as_field((dim,), xp.arange((0, grid.num_levels+1), dtype=dtype))
```
You should use `xp` from 
```from icon4py.model.common.settings import xp```
instead of numpy. This applies to the other tests as well